### PR TITLE
fix(trace): Handle 0 prefixed span ids

### PR DESCRIPTION
**Pull request by wmak**
- This improves the handling of the augment functions when span ids are prefixed by 0
- Fixes flakes with tests and usage where a span id starting with any number of `00` in them wouldn't be correctly associated
- Found an issue with the non useSpans usage of the endpoint, but skipping for now since the plan is to deprecate that usage

### CODE REVIEW

File: src/sentry/api/endpoints/organization_events_trace.py [hunk 1]
```diff
@@ -211,19 +211,16 @@ def load_performance_issues(self, light: bool, snuba_params: ParamsType) -> None
...
-                                    start = start_timestamp
-                                else:
-                                    start = min(start, start_timestamp)
-                                end_timestamp = float(event_span.get("precise.finish_ts"))
-                                if end is None:
-                                    end = end_timestamp
-                                else:
-                                    end = max(end, end_timestamp)
-                            except ValueError:
-                                pass
```
Comment (wmak) [inline src/sentry/api/endpoints/organization_events_trace.py:226]: This wasn't necessary and hiding bugs

File: tests/snuba/api/endpoints/test_organization_events_trace.py [hunk 2]
```diff
@@ -1612,20 +1618,6 @@ def test_indexed_spans_only_query_required_projects(self, mock_query_builder):
...
-            response = self.client_get(
-                data={"project": -1},
-            )
-        assert response.status_code == 200, response.content
-        trace_transaction = response.data["transactions"][0]
-        self.assert_trace_data(trace_transaction)
-        # We shouldn't have detailed fields here
-        assert "transaction.status" not in trace_transaction
-        assert "tags" not in trace_transaction
-        assert "measurements" not in trace_transaction
```
Comment (wmak) [inline tests/snuba/api/endpoints/test_organization_events_trace.py:1627]: Accidental commit in previous PR, this testcase is inherited

**Review by JoshFerge — COMMENTED** [review]
worth adding test cases where start/end are None?

File: src/sentry/api/endpoints/organization_events_trace.py [hunk 3]
```diff
@@ -562,8 +565,12 @@ def augment_transactions_with_spans(
         problem_project_map = {}
         issue_occurrences = []
         occurrence_spans = set()
-        error_spans = {e["trace.span"] for e in errors if e["trace.span"]}
-        projects = {e["project.id"] for e in errors if e["trace.span"]}
+        error_spans = set()
+        projects = set()
+        for error in errors:
```
Comment (JoshFerge) [inline src/sentry/api/endpoints/organization_events_trace.py:570]: why do we no longer have to check if the `trace.span` field exists in this loop before adding error spans / projects?

Comment (wmak) [reply src/sentry/api/endpoints/organization_events_trace.py:570]: woops good catch

**Comment by wmak** [comment]
> worth adding test cases where start/end are None?
Not really, if this is in reference to the try/catch the keys were just completely wrong so we werent attaching the perf issue. The ts fields are non-nullable so they can't be None

File: src/sentry/api/endpoints/organization_events_trace.py [hunk 4]
```diff
@@ -562,8 +565,13 @@ def augment_transactions_with_spans(
...
+            if "trace.span" in error:
+                error["trace.span"] = pad_span_id(error["trace.span"])
+                error_spans.add(error["trace.span"])
+            projects.add(error["project.id"])
```
Comment (shruthilayaj) [inline src/sentry/api/endpoints/organization_events_trace.py:574]: we were also only adding project id if trace.span, so does this also need to be nested under the if condition

**Review by shruthilayaj — APPROVED** [review]

File: src/sentry/api/endpoints/organization_events_trace.py (hunk 4 above)
Comment (wmak) [reply src/sentry/api/endpoints/organization_events_trace.py:574]: No i think that was just bad copy paste on my part lol
